integrations/FacebookPixel/browser.js

```diff
@@ -166,13 +178,22 @@ class FacebookPixel {
       const contents = [];
 
       if (products && Array.isArray(products)) {
-        products.forEach((product) => {
-          const productId = product.product_id;
+        products.forEach((product, index) => {
+          const productId =
+            get(product, "product_id") ||
+            get(product, "sku") ||
+            get(product, "id");
+          if (!isDefined(productId)) {
+            logger.error(
+              `Product id is required for product ${index}. Event not sent`
```

Comment:
This error message is misleading. The event is still sent but the product is skipped.